reference/debugging.md

```diff
@@ -0,0 +1,150 @@
+
+    ```console
+    $ kubectl get deployments
+    NAME         DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+    ambassador   3         3         3            3           1m
+    ```
+
+4. Check that the “desired” number of Pods equals the “current” or “available” number of pods. If they are **not** equal, check the status of the associated pods with the following command: `kubectl get pods -n ambassador`.
+5. Use the following command for details about the history of the Deployment: `kubectl describe -n ambassador deployment ambassador`
+
+    * Look for data in the “Replicas” field near the top of the output. For example: 
+        `Replicas: 3 desired | 3 updated | 3 total | 3 available | 0 unavailable`
+
+    * Look for data in the “Events” log field near the bottom of the output, which often displays data such as a fail image pull, RBAC issues, or a lack of cluster resources. For example:
+
+        ```console
+        Events:
+        Type    Reason              Age     From                      Message
+        ----    ------              ----    ----                      -------
+        Normal  ScalingReplicaSet    2m     deployment-controller      Scaled up replica set ambassador-85c4cf67b to 3
+        ```
+
+5. Additionally, use the following command to “describe” the individual pods: `kubectl describe pods -n ambassador <ambassador-pod-name>`
+
+    * Look for data in the “Status” field near the top of the output. For example, `Status: Running`
+
+    * Look for data in the “Events” field near the bottom of the output, as it will often show issues such as image pull failures, volume mount issues, and container crash loops. For example:
+        ```
+        Events:
+        Type    Reason                 Age   From                                                     Message
+        ----    ------                 ----  ----                                                     -------
+        Normal  Scheduled              4m    default-scheduler                                        Successfully assigned ambassador-85c4cf67b-4pfj2 to gke-ambassador-demo-default-pool-912378e5-dkxc
+        Normal  SuccessfulMountVolume  4m    kubelet, gke-ambassador-demo-default-pool-912378e5-dkxc  MountVolume.SetUp succeeded for volume "ambassador-token-tmk94"
+        Normal  Pulling                4m    kubelet, gke-ambassador-demo-default-pool-912378e5-dkxc  pulling image "quay.io/datawire/ambassador:0.40.0"
+        Normal  Pulled                 4m    kubelet, gke-ambassador-demo-default-pool-912378e5-dkxc  Successfully pulled image "quay.io/datawire/ambassador:0.40.0"
+        Normal  Created                4m    kubelet, gke-ambassador-demo-default-pool-912378e5-dkxc  Created container
+        Normal  Started                4m    kubelet, gke-ambassador-demo-default-pool-912378e5-dkxc  Started container
+        ```
+
+In both the Deployment pod and the individual pods, take the necessary action to address any discovered issues.
```

Comment:
I'm not sure what style is these days, but it's inconsistent that "Deployment" is uppercase but "pod" is lowercase.